### ISSUE TITLE
Don't unsubscribe MQTT Level 1 or above topics on nodered restart / close or at least add option

### ISSUE THREAD

**Issue by JsBergbau** [node-red/node-red#3452]
### Current Behavior

I use MQTT a lot. MQTT has with Level 1 the ability that no messages are lost and every message is processed. I process them via Node
![grafik](https://user-images.githubusercontent.com/37013344/154585011-3d9f0f37-541c-4756-bbe1-9a174ab916e5.png) and settings
![grafik](https://user-images.githubusercontent.com/37013344/154585112-6f0dca82-f5fc-4b29-b0b6-13c93c264ff8.png)

Multiple clients publish to Topic "queuedInput". If there is a power outage everything works fine, all messages are processed as they should.
If I reboot the system for updates nodered shuts down gracefully it unsubscribes that topic. That means all messages are lost in between the reboot time which is not what it should be.

You can see that all topics are unsubscribed on Node close in https://github.com/node-red/node-red/blob/fcf299401555e4ee02241999bb42fca80a1634b2/packages/node_modules/%40node-red/nodes/core/network/10-mqtt.js#L1066

### Expected Behavior

When using MQTT Level 1 and above and don't using clean session MQTT Node should not unsubscribe from those topics, so messages are queued in the MQTT broker as intended.

### Steps To Reproduce

Set up an MQTT-In Node with MQTT level 1, disabled clean session and client id. Publish data like an incrementing number every second and process it by writing to a file. 

Cut power of Node-RED system, wait a bit, then restart the system. Watch your file, all messages will be inside.

Now reboot the system, messages will be lost beginning with the reboot.

### Example flow

I think I've made the issue quite clear. If you really need full example codes, please tell, then I'll create some.

```
paste your flow here
```

### Environment

- Node-RED version: v2.1.4
- Node.js version: v12.22.10

The rest shouldn't matter. Even if this install is an older version, node close behaviour in the source code is the same as the current in https://github.com/node-red/node-red/blob/fcf299401555e4ee02241999bb42fca80a1634b2/packages/node_modules/%40node-red/nodes/core/network/10-mqtt.js#L1066

**Comment by knolleary** [comment]
This is more a feature request than a bug. There are good reasons for the current behaviour.

The question is whether we should have an option to disable the unsubscribe on close.

**Comment by dceejay** [comment]
If we do, it should probably only be enabled if the client is set to prevent orphan messages

**Comment by knolleary** [comment]
Yup - would need to differentiate between a node stopping and a node being deleted. In the latter case we definitely would still need to unsubscribe as the node won't be coming back to get the messages.

**Comment by dceejay** [comment]
And those with a random client if

**Comment by JsBergbau** [comment]
Thanks for the fast answers and thinking about this thoroughly.

If somebody uses message queuing, I expected that their broker is set accordingly. My relevant mosquitto config part is
```
max_queued_messages 100000
max_queued_bytes 0 #no limit, but is default

#Delete persistent data after 30 days if the client has not connected for that long
persistent_client_expiration 30d
```

If I comment that line https://github.com/node-red/node-red/blob/fcf299401555e4ee02241999bb42fca80a1634b2/packages/node_modules/%40node-red/nodes/core/network/10-mqtt.js#L1074
If I comment out this line, would that result in the desired behavior? Could any problems occur?

**Comment by Steve-Mcl** [comment]
@knolleary I have self assigned this one.

Are you happy for me to look at adding an option for this?

Also, do we think this would be suited to being per subscription, per node or per config/connection?

**Comment by knolleary** [comment]
@Steve-Mcl start with a single option per connection. I think doing it per subscription will get needlessly complicated.

**Comment by Steve-Mcl** [comment]
@knolleary 

After thinking about it, it probably makes most sense as a per connection thing.

This is a mockup - let me know if you are ok with the layout / working etc please...

![chrome_pCGB2yk2Dv](https://user-images.githubusercontent.com/44235289/158078702-2f637228-f61d-445a-b98b-4152a3122321.gif)

**Comment by knolleary** [comment]
We can iterate on the precising wording of the label in the PR, but I'd suggest something like "Automatically unsubscribe when disconnecting", with the default state being enabled (ie, the as-is behaviour).

**Comment by Steve-Mcl** [comment]
Ignore the referenced PR above - that was linked in error.

This issue/FR is still in progress

**Comment by aliblui** [comment]
Hi all,
sorry if this message is not relevant for the issue. It has been closed but I can't find reference in any release note and testing with 3.0.2 there is no option to change the behavior. At restart or deploy the session is unsubscribed. Is there any plan for release?

**Comment by knolleary** [comment]
Hi @aliblui 

This issue was closed automatically when the referenced PR was merged.

However checking @Steve-Mcl 's last comment, I can see the referenced PR was not related to this issue, so it should not have been closed. I'll reopen.

**Comment by flying7eleven** [comment]
Hi,

I was in need of this proposed solution. Since the last commend was from October 2022 I thought to give it a shot and try to implement it myself. I hope that it was okay for you @Steve-Mcl . If not, feel free to reject the PR. Otherwise any suggestions are welcome :)
